engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -825,10 +826,14 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
       case _ => false
     }
 
+    lazy val hasAnythingFailed = executionStore.values exists { x => x == Failed || x == Aborted }
+
+    lazy val workflowFailFastTrigger = workflowFailureMode == NoNewCalls && hasAnythingFailed
```

Comment:
@Horneth yes that's what I meant, thanks.  "most of this logic" was not a good way for me to describe that.  :smile: